assets/js/modules/analytics/components/common/PropertySelectIncludingGA4.test.js

```diff
@@ -0,0 +1,174 @@
+	dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'receiveGetWebDataStreams', { propertyID: propertyIDga4 } );
+};
+
+const setupEmptyRegistry = ( { dispatch } ) => {
+	dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetProperties( [], { accountID } );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties( [], { accountID } );
+	dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'getProperties', [ accountID ] );
+};
+
+describe( 'PropertySelectIncludingGA4', () => {
+	it( 'should render an option for each analytics property of the currently selected account.', async () => {
+		const { getAllByRole } = render( <PropertySelect />, { setupRegistry } );
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		// Note: we do length + 1 here because there should also be an item for
+		// "Set up a new property".
+		expect( listItems ).toHaveLength( propertiesUA.length + propertiesGA4.length + 1 );
+	} );
+
+	it( 'should not render in the absence of an valid account ID.', async () => {
+		const { container, registry } = render( <PropertySelect />, {
+			setupRegistry,
+		} );
+
+		// A valid accountID is provided, so ensure it is not currently disabled.
+		const selectWrapper = container.querySelector( '.googlesitekit-analytics__select-property' );
+		const selectedText = container.querySelector( '.mdc-select__selected-text' );
+		expect( selectWrapper ).not.toHaveClass( 'mdc-select--disabled' );
+		expect( selectedText ).not.toHaveAttribute( 'aria-disabled', 'true' );
+
+		act( () => {
+			registry.dispatch( MODULES_ANALYTICS ).setAccountID( ACCOUNT_CREATE );
+			registry.dispatch( MODULES_ANALYTICS ).finishResolution( 'getProperties', [ ACCOUNT_CREATE ] );
+			registry.dispatch( MODULES_ANALYTICS_4 ).finishResolution( 'getProperties', [ ACCOUNT_CREATE ] );
+		} );
+		// ACCOUNT_CREATE is an invalid accountID (but valid selection), so ensure the select is not rendered
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( 'should render a select box with only an option to create a new property if no properties are available.', async () => {
+		const { getAllByRole } = render( <PropertySelect />, { setupRegistry: setupEmptyRegistry } );
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		expect( listItems ).toHaveLength( 1 );
+		expect( listItems[ 0 ].textContent ).toMatch( /set up a new property/i );
+	} );
+
+	it( 'should change between UA and GA4 properties', async () => {
+		const { getAllByRole, container, registry } = render( <PropertySelect />, { setupRegistry } );
```

Comment:
it's a long test but I combined a few together.  It gives me confidence anyway that the many parts of our state are being changed correctly!